### ISSUE TITLE
'ExtractMethodCodeRefactoringProvider' encountered an error

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#40555]
```
System.InvalidOperationException : Sequence contains no elements
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source)
   at Microsoft.CodeAnalysis.ExtractMethod.InsertionPoint.ComputeContextNode()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at Microsoft.CodeAnalysis.ExtractMethod.InsertionPoint.GetContext()
   at async Microsoft.CodeAnalysis.ExtractMethod.MethodExtractor.CodeGenerator`3.GenerateAsync[TStatement,TExpression,TNodeUnderContainer](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractMethod.MethodExtractor.ExtractMethodAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.ExtractMethod.AbstractExtractMethodService`3.ExtractMethodAsync[TValidator,TExtractor,TResult](<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.ExtractMethod.ExtractMethodCodeRefactoringProvider.ExtractLocalFunction(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.ExtractMethod.ExtractMethodCodeRefactoringProvider.GetCodeActionsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.ExtractMethod.ExtractMethodCodeRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

**Comment by allisonchou** [comment]
@AlekseyTs Do you perhaps have a repro for this?

**Comment by AlekseyTs** [comment]
> Do you perhaps have a repro for this?

Unfortunately, no. A message popped up in VS, the exception information was in the log file.

**Comment by mavasani** [comment]
I am seeing this every few minutes on the latest VS2019 16.5 Preview builds, I think we should fix this for 16.5. I will try to see if I can get a repro.

**Comment by mavasani** [comment]
Repro steps:

1. git clone https://github.com/mavasani/roslyn-analyzers.git
2. cd roslyn-analyzers
3. git checkout origin/useCorrectCancellationToken
4. restore
5. RoslynAnalyzers.sln
6. Open source file `src\NetAnalyzers\Core\Microsoft.NetCore.Analyzers\Runtime\DisposableFieldsShouldBeDisposed.cs`
7. Ctrl + F and find "CancellationToken" and keep hitting enter to move through all the search results. You should see the exception when you reach the local function with CancellationToken parameter.
